src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2157,6 +2164,21 @@ private static void AddEntryPointCandidates(
             ArrayBuilder<MethodSymbol> entryPointCandidates, IEnumerable<Symbol> members)
         {
             foreach (var member in members)
+            {
+                if (member.GetIsNewExtensionMember())
```

Comment:
tbh, i struggle finding any reason an extension member should ever be the entrypoint.  why not just diallow it? :)

Reply:
Ah ok.  That makes sense.  Thanks!